### PR TITLE
HL-99 | Deleting draft applications

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
## Description :sparkles:
- Implement a shared Modal component
- Implement the deleting logic in the form actions custom hook
- Add the deleting functionality to all steps

## Testing :alembic:
- From the application list, click on the edit button of one of the draft applications
- At the bottom of the page, click on the delete button
- A confirmation modal should open
- By confirming, the application should be deleted and you'll be redirected to the home page
- A success toast will appear to confirm that the application has been successfully deleted

## Screenshots :camera_flash:
image

### CODE REVIEW

File: frontend/benefit/applicant/src/hooks/useDeleteApplicationQuery.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { ErrorData } from '../types/common';
+
+const useDeleteApplicationQuery = (): UseMutationResult<
+  null,
+  AxiosError<ErrorData>,
+  string
+> => {
+  const { axios, handleResponse } = useBackendAPI();
+  const queryClient = useQueryClient();
+  return useMutation<null, AxiosError<ErrorData>, string>(
```
Comment (nenonja) [inline frontend/benefit/applicant/src/hooks/useDeleteApplicationQuery.ts:15]: type generics are probably redundant here

File: frontend/benefit/applicant/src/hooks/useFormActions.tsx [hunk 2]
```diff
@@ -184,10 +197,29 @@ const useFormActions = (application: Application): FormActions => {
     return undefined;
   };
 
+  const onDelete = async (id: string): Promise<void> => {
+    try {
+      await deleteApplication(id);
```
Comment (nenonja) [inline frontend/benefit/applicant/src/hooks/useFormActions.tsx:202]: I probably would not use `mutateAsync` but `mutate` + `onSuccess`-callback instead: 
https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutate-or-mutateasync
https://tkdodo.eu/blog/mastering-mutations-in-react-query#some-callbacks-might-not-fire


> Do UI related things like redirects or showing toast notifications in mutate callbacks. If the user navigated away from the current screen before the mutation finished, those will purposefully not fire.

File: frontend/shared/src/components/modal/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { Button, Dialog, DialogVariant } from 'hds-react';
```
Comment (nenonja) [inline frontend/shared/src/components/modal/Modal.tsx:1]: Good job! Now other projects can use this as well 👍

File: frontend/shared/src/components/modal/__tests__/Modal.test.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  const getComponent = (props: Partial<ModalProps> = {}): RenderResult =>
+    render(<Modal {...initialProps} {...props} />);
+
+  it('should render with no accessibility violations', async () => {
+    const { container } = getComponent();
+    const results = await axe(container);
+
+    expect(results).toHaveNoViolations();
+  });
```
Comment (nenonja) [inline frontend/shared/src/components/modal/__tests__/Modal.test.tsx:24]: Probabhy there could be another test to check that the `handleSubmit` and `handleToggle` gets called when clicking the button

**Review by nenonja — COMMENTED** [review]
Looks good generally! Added couple of comments

File: frontend/benefit/applicant/src/components/applications/forms/application/stepperActions/StepperActions.tsx [hunk 5]
```diff
@@ -18,54 +20,80 @@ const StepperActions: React.FC<StepperActionsProps> = ({
...
+        <$GridCell $colSpan={10} $colStart={2} justifySelf="center">
           <Button
             theme="black"
-            variant="secondary"
-            iconLeft={<IconArrowLeft />}
-            onClick={handleBack}
+            variant="supplementary"
+            iconLeft={<IconCross />}
+            onClick={() => setIsConfirmationModalOpen(true)}
+            disabled={!handleDelete}
```
[the remark this answers is not in the document]
Comment (Nizar-Rahme) [reply frontend/benefit/applicant/src/components/applications/forms/application/stepperActions/StepperActions.tsx]: Good point! Fixed.

File: frontend/benefit/applicant/src/components/applications/forms/application/stepperActions/StepperActions.tsx [hunk 6]
```diff
@@ -18,54 +20,80 @@ const StepperActions: React.FC<StepperActionsProps> = ({
...
-      </$GridCell>
-      <$GridCell $colSpan={3} justifySelf="end">
-        <Button
-          theme="coat"
-          disabled={disabledNext}
-          iconRight={!lastStep ? <IconArrowRight /> : null}
-          onClick={handleSubmit}
+        </$GridCell>
+      </$Grid>
+      {isConfirmationModalOpen && (
```
[the remark this answers is not in the document]
Comment (Nizar-Rahme) [reply frontend/benefit/applicant/src/components/applications/forms/application/stepperActions/StepperActions.tsx]: Fixed

File: frontend/shared/src/components/modal/__tests__/Modal.test.tsx (hunk 4 above)
Comment (Nizar-Rahme) [reply frontend/shared/src/components/modal/__tests__/Modal.test.tsx:28]: Tests added.

File: frontend/benefit/applicant/src/hooks/useFormActions.tsx (hunk 2 above)
Comment (Nizar-Rahme) [reply frontend/benefit/applicant/src/hooks/useFormActions.tsx]: Those articles 💎 
Fixed

**Review by nenonja — APPROVED** [review]
Looks good now!

**Pull request merged**